CHANGELOG.md

```diff
@@ -3,6 +3,333 @@
+* ([#2257](https://github.com/serenity-rs/serenity/pull/2257)) Add support for multiple event handlers by replacing the `event_handler` and `raw_event_handler` fields with pluralized `event_handlers` and `raw_event_handlers` in the following structs:
+    - `ShardManagerOptions`
+    - `ShardQueuer`
+    - `ShardRunner`
+    - `ShardRunnerOptions`
+    - `ClientBuilder`
+* ([#2273](https://github.com/serenity-rs/serenity/pull/2273), [#2367](https://github.com/serenity-rs/serenity/pull/2367)) Add events `ReactionRemoveEmoji` and `GuildAuditLogEntryCreate`.
+* ([#2276](https://github.com/serenity-rs/serenity/pull/2276)) Add support for automod regex patterns.
+* ([#2297](https://github.com/serenity-rs/serenity/pull/2297)) Add the `serenity::all` module, which re-exports most public items in the crate.
+* ([#2336](https://github.com/serenity-rs/serenity/pull/2336)) Add a `CreateButton::custom_id` method.
+* ([#2369](https://github.com/serenity-rs/serenity/pull/2369)) Add support for editing a guild's MFA level using `{GuildId, Guild, PartialGuild}::edit_mfa_level`.
+* ([#2391](https://github.com/serenity-rs/serenity/pull/2391)) Add attachments support to the `EditWebhookMessage` endpoint by adding a `new_attachments` parameter to `Http::edit_webhook_message`, as well as the following methods to the `EditWebhookMessage` builder:
+    - `attachment`
+    - `add_existing_attachment`
+    - `remove_existing_attachment`
+
+#### Changed
+
+* ([#1896](https://github.com/serenity-rs/serenity/pull/1896)) `Request::body_ref` now returns `Option<&T>` instead of `&Option<&T>`.
+* ([#1897](https://github.com/serenity-rs/serenity/pull/1897), [#2350](https://github.com/serenity-rs/serenity/pull/2350)) `Typing::stop` now returns `bool` instead of `Option<()>`. Also, `Typing::start` and any wrapper methods are now infallible.
+* ([#1922](https://github.com/serenity-rs/serenity/pull/1922), [#1940](https://github.com/serenity-rs/serenity/pull/1940), [#2090](https://github.com/serenity-rs/serenity/pull/2090)) The following methods are no longer `async`:
+    - `ChannelId::name`
+    - `Context::*`
+    - `Guild::{members_starting_with, members_containing, members_username_containing, members_nick_containing}`
+    - `Guild::default_channel`
+    - `PartialGuild::greater_member_hierarchy`
+    - `ShardManager::new`
+    - `UserId::to_user_cached`
+* ([#1929](https://github.com/serenity-rs/serenity/pull/1929)) Unbox the `Error::Http` variant.
+* ([#1930](https://github.com/serenity-rs/serenity/pull/1930)) Change `Id` types to wrap `NonZeroU64` instead of `u64`. The new API is as follows:
+    - `ExampleId(12345)` -> `ExampleId::new(12345)`
+    - `example_id.0` -> `example_id.get()`
+    - `example_id.as_u64()` -> `example_id.as_inner()`
```

Comment:
This is a pretty major change for users hard coding IDs in their code, especially using ID `0` as a dummy value, so should be given it's own section.

Reply:
Done. Also realized that `.get` and `.as_inner` were swapped.